### PR TITLE
Track the task preprocessor layer as part of model

### PR DESCRIPTION
**Pull request by mattdangerw**
Remove it from model.layers, so it does not show up in summary

### CODE REVIEW

**Comment by mattdangerw** [comment]
A few other ways we could do this...

 - Use the `layer_range` argument `model.summary` and rely on the fact that `preproccessor` is currently always last in the layer list.
 - Use the `layer_range` argument as above and override `layers` for force preprocessor to be last (safer).

For now, I am just removing `preprocessor` from `layers`, as I couldn't think of a practical reason it needed to be there. This is the least code.

**Comment by mattdangerw** [comment]
Makes for a fairly nice/concise export flow! https://colab.sandbox.google.com/gist/mattdangerw/0a1ae7806b23e1c4712ab1aa4cc6499f/export-recipe.ipynb

File: keras_nlp/models/task.py [hunk 1]
```diff
@@ -30,9 +29,9 @@ class Task(PipelineModel):
     """Base class for Task models."""
 
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
         self._backbone = None
         self._preprocessor = None
+        super().__init__(*args, **kwargs)
```
Comment (chenmoneygithub) [inline keras_nlp/models/task.py:34]: curious - Why do we need to move the order here?

File: keras_nlp/models/task.py [hunk 2]
```diff
@@ -178,6 +176,14 @@ def from_preset(calling_cls, *args, **kwargs):
                 preset_names='", "'.join(cls.presets),
             )(cls.from_preset.__func__)
 
+    @property
+    def layers(self):
+        # Remove preprocessor from layers so it does not show up in the summary.
```
Comment (chenmoneygithub) [inline keras_nlp/models/task.py:181]: Should we write it in docstring for better visibility?

**Review by chenmoneygithub — APPROVED** [review]
Thanks Matt! LGTM, 2 minor comments.

File: keras_nlp/models/task.py (hunk 1 above)
Comment (jbischof) [reply keras_nlp/models/task.py:34]: Should I swap this for KerasCV as well? Not totally sure I understand the problem.

**Pull request merged**